bpf/libbpf/libbpf_api.h

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "libbpf.h"
+#include <linux/limits.h>
+#include <net/if.h>
+#include <bpf.h>
+#include <stdlib.h>
+#include <errno.h>
+
+struct bpf_object* bpf_obj_open_load(char *filename) {
+	errno = 0;
+	return bpf_object__open(filename);
```

Comment:
Are you sure that `bpf_object__open` sets `errno` in all failure cases?  For a case where the file doesn't exist, I'd expect errno to be set as a side effect of `bpf_object__open` calling `open` under the covers but for other errors, I'm not sure if `bpf_object__open` would set `errno` or if it's return a fake error pointer.

Sadly, in C, it's hard to be sure without code reading/documentation.

Reply:
I overlooked the libbpf version. I had libbpf version 0.5 in my laptop and that had proper error handling. Guess will be better to move to libbpf v0.5.0. WDYT?

Reply:
Tried with v0.5.0, but looks like relying on errno set by libbpf may not help

Reply:
libbpf's convention is to return `0` on success and `-errno` (in 0.4.0) changed to `+errno` later on. It does not set `errno`. So as @sridhartigera and I discussed, imo the solution is to call into libbpf and than set the `errno`. It is not clear whether you need to return a negative return value. From @fasaxc 's description it seems like you do not need to.

I'd say that to be compatible with future upgrades of libbpf, you should do

```
void myfunc(void)
{
    int ret = libbpf_some_functio();
    errno = ret >= 0 ? : -ret;
}
```